kedro/extras/datasets/pickle/pickle_dataset.py

```diff
@@ -203,14 +203,16 @@ def _load(self) -> Any:
         load_path = get_filepath_str(self._get_load_path(), self._protocol)
 
         with self._fs.open(load_path, **self._fs_open_args_load) as fs_file:
-            return self._backend.load(fs_file, **self._load_args)  # type: ignore
+            imported_backend = importlib.import_module(self._backend)
```

Comment:
Optional, very minor: naming this the same to what we use in `__init__` made me instantly think "why not just save it on the instance as an attribute", which would make us end up in the same bad spot we were before. I think it's possible that other people would have the same thought reading the code. Maybe we can use a different variable name, `pickle_lib`, `pickling_lib`, `pickle_backend` etc.

Reply:
I deliberately made it `imported_backend` (rather than `backend`) exactly for this reason 🙂  Don't know whether you misread it or whether you think the difference in variable names is not clear enough yet?